src/context/cardContext.tsx

```diff
@@ -0,0 +1,78 @@
+import  { createContext, useContext, useState, ReactNode, useEffect } from "react";
+
+interface CartItem {
+  id: string;
+  title: string;
+  price: number;
+  quantity: number;
+  image: string;
+}
+
+interface CartContextType {
+  cartItems: CartItem[];
+  addToCart: (item: CartItem) => void;
+  removeFromCart: (id: string) => void;
+  updateQuantity: (id: string, quantity: number) => void;
+  clearCart: () => void;
+}
+
+const CartContext = createContext<CartContextType | undefined>(undefined);
+
+export const CartProvider = ({ children }: { children: ReactNode }) => {
+  const [cartItems, setCartItems] = useState<CartItem[]>(() => {
+    const savedCart = localStorage.getItem("cart");
+    return savedCart ? JSON.parse(savedCart) : [];
+  });
```

Comment:
A better way would be to create separate variable and pass it as default value rather then passing a anonymous function

```
 const savedCart = JSON.parse(localStorage.getItem('cart')) || [];
const [cartItems, setCartItems] = useState<CartItem[]>([]);
```